orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt

```diff
@@ -144,4 +147,39 @@ class CompleteStageHandler(
       this.execution = repository.retrieve(this.execution.type, this.execution.id)
     }
   }
+
+  /**
+   * Plan any outstanding synthetic on failure stages.
+   */
+  private fun Stage.planOnFailureStages(): List<Stage> {
+    builder().let { builder ->
+      val previouslyPlannedAfterStageNames = afterStages().map { it.name }
```

Comment:
Is there a better way to avoid adding stages that were planned/inserted previously?